zk/conn.go

```diff
@@ -984,6 +968,9 @@ func (c *Conn) Children(path string) ([]string, *Stat, error) {
 
 	res := &getChildren2Response{}
 	_, err := c.request(opGetChildren2, &getChildren2Request{Path: path, Watch: false}, res, nil)
+	if err != nil && err == ErrConnectionClosed {
+		return nil, nil, err
```

Comment:
not sure why we don't want to return an error to the caller if the connection is closed.

could you explain?

Reply:
Yeah im not super sure on this one as it comes from the mesos fork. I will consider this and look more into why this was done. it might be unwanted.

Reply:
updated with removing commits around this connection error handling. the changes are more concise and do not change the returning of errors. I think the racy comments around queuing requests from the commits were valid but we can solve this with doing a better coordination of send/recv routines.